python/mlx/optimizers.py

```diff
@@ -105,7 +118,17 @@ def apply_single(
             return parameter - self.learning_rate * gradient
 
         v = state.get("v", mx.zeros_like(gradient))
-        v = self.momentum * v + (1 - self.momentum) * gradient
+
+        if self.weight_decay != 0:
+            gradient += self.weight_decay * parameter
+
+        buf = self.momentum * v + (1 - self.dampening) * gradient
+
+        if self.nesterov:
+            v += self.momentum * buf
+        else:
+            v = buf
```

Comment:
I don't think this is correct. You don't want to update `v` in the nesterov case.

I think it should be something like:

```
v = self.momentum * v
if dampening > 0:
  v += (1 - dampening) * gradient
else:
  v += gradient

if self.nesterov
  update = gradient  + self.momentum * v
else:
  update = v
state["v"] = v
return parameter - self.learning_rate * update
```

Reply:
My bad. I will update this.

Also, in PyTorch's SGD constructor, they have this check:
```
if nesterov and (momentum <= 0 or dampening != 0):
    raise ValueError("Nesterov momentum requires a momentum and zero dampening")
```
https://pytorch.org/docs/stable/_modules/torch/optim/sgd.html#SGD
Should this be added since I'm adding the `dampening` parameter?

Reply:
Yea, good call, I think it's worth adding.